platform/osgi/platform-osgi-serviceflag/src/main/java/org/codice/ddf/platform/serviceflag/inputtransformer/InputTransformerServiceFlag.java

```diff
@@ -0,0 +1,196 @@
+    this.transformerCheckPeriodMillis = DEFAULT_TRANSFORMER_CHECK_PERIOD_MILLIS;
+    this.transformerWaitTimeoutMillis = getTransformerWaitTimeout();
+    executorService.submit(
+        () -> waitForInputTransformers(FrameworkUtil.getBundle(this.getClass())));
+  }
+
+  @VisibleForTesting
+  InputTransformerServiceFlag(
+      InputTransformerIds inputTransformerIds,
+      List<ServiceReference<InputTransformer>> inputTransformers,
+      Bundle bundle,
+      long transformerCheckPeriodMillis,
+      long transformerWaitTimeoutMillis) {
+    this.inputTransformerIds = inputTransformerIds;
+    this.inputTransformers = inputTransformers;
+    this.transformerCheckPeriodMillis = transformerCheckPeriodMillis;
+    this.transformerWaitTimeoutMillis = transformerWaitTimeoutMillis;
+    waitForInputTransformers(bundle);
+  }
+
+  /**
+   * Waits up to a configured ({@code org.codice.ddf.cdm.transformerWaitTimeoutSeconds} system
+   * property) amount of time for the configured {@link InputTransformer} services to be available.
+   */
+  @SuppressWarnings({
+    "squid:S1481" /* Sonar thinks variables are unused */,
+    "squid:S1854" /* Useless assignment for variable Sonar thinks is unused */
+  })
+  private void waitForInputTransformers(Bundle bundle) {
+    Set<String> configuredInputTransformers = inputTransformerIds.getIds();
+    if (configuredInputTransformers.isEmpty()) {
+      publishService(bundle);
+      destroy();
+      return;
+    }
+
+    Set<String> requiredInputTransformers = new HashSet<>(configuredInputTransformers);
+    LOGGER.trace(
+        "Beginning to wait for InputTransformers for {} seconds",
+        TimeUnit.MILLISECONDS.toSeconds(transformerWaitTimeoutMillis));
+
+    Failsafe.with(
+            new RetryPolicy()
+                .withMaxDuration(transformerWaitTimeoutMillis, TimeUnit.MILLISECONDS)
+                .retryWhen(false)
+                .withDelay(transformerCheckPeriodMillis, TimeUnit.MILLISECONDS))
+        .withFallback(
+            () -> {
+              throw new IllegalArgumentException(
+                  String.format(
+                      "Not all expected Input Transformers were found within %s seconds. Expected: %s, but didn't find: %s. "
+                          + "CDMs will not be able to start without all required InputTransformers. Either remove the un-found "
+                          + "IDs from the InputTransformer ID JSON files under %s, or ensure the bundles providing "
+                          + "the missing InputTransformers are started.",
+                      TimeUnit.MILLISECONDS.toSeconds(transformerWaitTimeoutMillis),
+                      configuredInputTransformers,
+                      requiredInputTransformers,
+                      inputTransformerIds.getTransformerPath()));
+            })
+        .get(() -> checkForInputTransformers(requiredInputTransformers));
+    publishService(bundle);
```

Comment:
❗️ We should be monitoring when input transformers go down and unregistering this service accordingly. This will ensure the system is refreshing properly. I'm not sure if the services stay registered when the bundle is stopped, if not, we should be unregistering then too.

Reply:
I'm still on the fence about this. If many services end up relying on this in the future, is it reasonable for and Admin to put dependent parts (potentially critical parts) of the system in grace period because they disable an InputTransformer they don't want?